edx_gea/gea.py

```diff
@@ -1,40 +1,118 @@
-"""TO-DO: Write a description of what this XBlock is."""
+"""GEA (Grade External Activity). An XBlock used to grade external activities.
 
-import pkg_resources
+All kind of student activities can't be achieved inside edx-platform. Therefore a teacher
+will only describe the activity in the lms and add a GEA XBlock to grade it.
+"""
+
+import csv
+
+from django.contrib.auth.models import User
+from django.core.exceptions import PermissionDenied
+from django.utils.translation import ugettext_lazy, ugettext as _
+
+from webob.response import Response
 
 from xblock.core import XBlock
-from xblock.fields import Scope, String
+from xblock.fields import Scope, String, Integer
 from xblock.fragment import Fragment
+from xblockutils.studio_editable import StudioEditableXBlockMixin
+from xblockutils.resources import ResourceLoader
+
+from .forms import UploadAssessmentFileForm
+from .gea_assessment import GeaAssessment, Score
+
+
+loader = ResourceLoader(__name__)
 
-from django.utils.translation import ugettext as _
+class GradeExternalActivityXBlock(XBlock, StudioEditableXBlockMixin):
+    """XBlock to grade external activities."""
 
-class GradeExternalActivityXBlock(XBlock):
-    """
-    """
+    has_score = True #: This flags the XBlock as a problem.
+    icon_class = 'problem'
 
     display_name = String(
-        default=_('External Activity'), scope=Scope.settings,
+        default=ugettext_lazy('External Activity Grader'), scope=Scope.settings,
         help="This name appears in the horizontal navigation at the top of "
              "the page."
     )
 
-    def resource_string(self, path):
-        """Handy helper for getting resources from our kit."""
+    max_score = Integer(
+        display_name=ugettext_lazy("Maximum grade"),
+        help=(ugettext_lazy("Maximum grade of the external activity.")),
+        default=10,
+        scope=Scope.settings
+    )
 
-        data = pkg_resources.resource_string(__name__, path)
-        return data.decode("utf8")
+    editable_fields = ('max_score',)
 
     def student_view(self, context=None):
+        """Display the student assessment (score and comment).
+
+        Note:
+            The LMS Runtime only call the XBlock.student_view. In order to get two different
+            views, whereas the runtime user is a staff member or a student, we check inside the student_view
+            if the user is a staff and call the staff_view accordingly.
```

Comment:
"In order to get two different views, depending on the runtime user being a staff member or a student, ..."